### PR TITLE
DEVELOPER-5093 - Fix Unit Tests

### PR DESCRIPTION
**Pull request by KamiQuasi**
### JIRA Issue Link
* https://issues.jboss.org/browse/DEVELOPER-5093

### Verification Process
TBD

### CODE REVIEW

**Comment by RubyGemTSL** [comment]
@KamiQuasi this is still failing to build.

**Comment by robpblake** [comment]
retest this please

**Comment by robpblake** [comment]
@KamiQuasi Drupal is failing to boot with:

`Rendering Complete, saving .css file...
Wrote CSS to /var/www/drupal/web/themes/custom/rhdp/rhd-frontend/styles/rhd.css
Wrote 1 CSS files to /var/www/drupal/web/themes/custom/rhdp/rhd-frontend/styles/
Rendering Complete, saving .css file...
Wrote CSS to /var/www/drupal/web/themes/custom/rhdp/rhd-frontend/rhd.css
Wrote 1 CSS files to /var/www/drupal/web/themes/custom/rhdp/rhd-frontend/
{ Error: Cannot find module 'caniuse-db/data.json'
    at Function.Module._resolveFilename (module.js:547:15)
    at Function.Module._load (module.js:474:25)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/var/www/drupal/web/themes/custom/rhdp/rhd-frontend/node_modules/caniuse-api/node_modules/browserslist/index.js:5:15)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3) code: 'MODULE_NOT_FOUND' }
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! rhd-frontend@1.1.101 styles: `node-sass src/styles/ -o styles/ && node-sass src/styles/ -o ./ --output-style compressed && postcss -o rhd.min.css rhd.css`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the rhd-frontend@1.1.101 styles script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2018-09-18T09_43_29_813Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! rhd-frontend@1.1.101 build: `npm run scripts && npm run styles`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the rhd-frontend@1.1.101 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2018-09-18T09_43_29_827Z-debug.log`

**Comment by LightGuard** [comment]
Latest error:

```
14:54:41 
14:54:41 > rhd-frontend@1.1.101 test:docker /home/unit/rhd-frontend
14:54:41 > npm run build && karma start src/tests/unit/config/karma.docker.conf.js
14:54:41 
14:54:41 
14:54:41 > rhd-frontend@1.1.101 build /home/unit/rhd-frontend
14:54:41 > npm run scripts && npm run styles
14:54:41 
14:54:41 
14:54:41 > rhd-frontend@1.1.101 scripts /home/unit/rhd-frontend
14:54:41 > tsc && uglifyjs rhd.js -c -m -o rhd.min.js
14:54:41 
14:54:41 sh: tsc: command not found
14:54:41 npm ERR! file sh
14:54:41 npm ERR! code ELIFECYCLE
14:54:41 npm ERR! errno ENOENT
14:54:41 npm ERR! syscall spawn
14:54:41 npm ERR! rhd-frontend@1.1.101 scripts: `tsc && uglifyjs rhd.js -c -m -o rhd.min.js`
14:54:41 npm ERR! spawn ENOENT
14:54:41 npm ERR! 
14:54:41 npm ERR! Failed at the rhd-frontend@1.1.101 scripts script.
14:54:41 npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
14:54:41 npm WARN Local package.json exists, but node_modules missing, did you mean to install?
14:54:41 
14:54:41 npm ERR! A complete log of this run can be found in:
14:54:41 npm ERR!     /home/unit/.npm/_logs/2018-09-18T14_54_31_909Z-debug.log
14:54:41 npm ERR! code ELIFECYCLE
14:54:41 npm ERR! errno 1
14:54:41 npm ERR! rhd-frontend@1.1.101 build: `npm run scripts && npm run styles`
14:54:41 npm ERR! Exit status 1
14:54:41 npm ERR! 
14:54:41 npm ERR! Failed at the rhd-frontend@1.1.101 build script.
14:54:41 npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
14:54:41 npm WARN Local package.json exists, but node_modules missing, did you mean to install?
14:54:41 
14:54:41 npm ERR! A complete log of this run can be found in:
14:54:41 npm ERR!     /home/unit/.npm/_logs/2018-09-18T14_54_31_920Z-debug.log
14:54:41 npm ERR! code ELIFECYCLE
14:54:41 npm ERR! errno 1
14:54:41 npm ERR! rhd-frontend@1.1.101 test:docker: `npm run build && karma start src/tests/unit/config/karma.docker.conf.js`
14:54:41 npm ERR! Exit status 1
14:54:41 npm ERR! 
14:54:41 npm ERR! Failed at the rhd-frontend@1.1.101 test:docker script.
14:54:41 npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
14:54:41 npm WARN Local package.json exists, but node_modules missing, did you mean to install?
```

**Comment by RubyGemTSL** [comment]
retest this please

**Comment by RubyGemTSL** [comment]
@robpblake this PR is killing me. I finally have them running in docker and then the build fails :( can you help with the error?

`[1BError: Docker container 'rhdpr2611_drupal_1' does not appear to be running. Please use 'docker logs -f rhdpr2611_drupal_1' to investigate any start-up failures.`

**Comment by RubyGemTSL** [comment]
I see why now I checked the logs.. fixing @KamiQuasi

**Comment by RubyGemTSL** [comment]
@KamiQuasi PR for you with hopefully my last update :)

**Comment by KamiQuasi** [comment]
retest this please

**Comment by RubyGemTSL** [comment]
@KamiQuasi this should fix the package.json file missing error. However there are some conflicts with master that I'd rather not fix in case I mess anything up. Can you address?

**Comment by LightGuard** [comment]
retest this please

**Comment by LightGuard** [comment]
Most recent End to end failures:

```
AssertionError: expected NaN to be at most NaN
    at Context.<anonymous> (/home/e2e/developers.redhat.com/_tests/e2e/specs/rhd-search.spec.js:35:64)
    at new Promise (<anonymous>)
    at new F (/home/e2e/node_modules/core-js/library/modules/_export.js:36:28)
```

**Comment by LightGuard** [comment]
Latest failures for unit tests, at least this seems to be the root cause

```
19:14:39 > developers.redhat.com@1.0.0 test /home/unit/developers.redhat.com
19:14:39 > karma start _docker/drupal/drupal-filesystem/web/themes/custom/rhdp/rhd-frontend/src/tests/unit/config/karma.conf.js
19:14:39 
19:14:40 01 10 2018 19:14:29.449:WARN [watcher]: Pattern "/home/unit/developers.redhat.com/_docker/drupal/drupal-filesystem/web/themes/custom/rhdp/rhd-frontend/rhd.min.js" does not match any file.
19:14:40 01 10 2018 19:14:29.480:INFO [karma]: Karma v3.0.0 server started at http://0.0.0.0:9876/
19:14:40 01 10 2018 19:14:29.481:INFO [launcher]: Launching browser ChromeHeadless with unlimited concurrency
19:14:40 01 10 2018 19:14:29.487:INFO [launcher]: Starting browser ChromeHeadless
19:14:40 01 10 2018 19:14:30.001:INFO [HeadlessChrome 0.0.0 (Linux 0.0.0)]: Connected on socket bcDCVTxaJEc2TRcUAAAA with id 81963445
19:14:40 01 10 2018 19:14:30.164:WARN [web-server]: 404: /@rhd/dp-category-list/dp-category-item-list
19:14:40 01 10 2018 19:14:30.167:WARN [web-server]: 404: /@rhd/dp-category-list/dp-category-item
19:14:40 01 10 2018 19:14:30.172:WARN [web-server]: 404: /@rhd/dp-category-list/dp-category-list
19:14:40 01 10 2018 19:14:30.172:WARN [web-server]: 404: /@rhd/dp-category-list/dp-category
19:14:40 01 10 2018 19:14:30.172:WARN [web-server]: 404: /@rhd/dp-category-list/dp-product-short-teaser
19:14:40 01 10 2018 19:14:30.172:WARN [web-server]: 404: /@rhd/rhdp-search/rhdp-search-app
19:14:40 01 10 2018 19:14:30.173:WARN [web-server]: 404: /@rhd/rhdp-search/rhdp-search-box
19:14:40 01 10 2018 19:14:30.173:WARN [web-server]: 404: /@rhd/rhdp-search/rhdp-search-filter-group
19:14:40 01 10 2018 19:14:30.175:WARN [web-server]: 404: /@rhd/rhdp-search/rhdp-search-filter-item
19:14:40 01 10 2018 19:14:30.177:WARN [web-server]: 404: /@rhd/rhdp-search/rhdp-search-filters
19:14:40 01 10 2018 19:14:30.177:WARN [web-server]: 404: /@rhd/rhdp-search/rhdp-search-onebox
19:14:40 01 10 2018 19:14:30.177:WARN [web-server]: 404: /@rhd/rhdp-search/rhdp-search-query
19:14:40 01 10 2018 19:14:30.178:WARN [web-server]: 404: /@rhd/rhdp-search/rhdp-search-result-count
19:14:40 01 10 2018 19:14:30.178:WARN [web-server]: 404: /@rhd/rhdp-search/rhdp-search-result
19:14:40 01 10 2018 19:14:30.180:WARN [web-server]: 404: /@rhd/rhdp-search/rhdp-search-results
19:14:40 01 10 2018 19:14:30.181:WARN [web-server]: 404: /@rhd/rhdp-search/rhdp-search-sort-page
19:14:40 01 10 2018 19:14:30.181:WARN [web-server]: 404: /@rhd/rhdp-search/rhdp-search-url
19:14:40 01 10 2018 19:14:30.181:WARN [web-server]: 404: /@rhd/rhdp-downloads/rhdp-downloads-all-item
19:14:40 01 10 2018 19:14:30.182:WARN [web-server]: 404: /@rhd/rhdp-downloads/rhdp-downloads-all
19:14:40 01 10 2018 19:14:30.182:WARN [web-server]: 404: /@rhd/rhdp-downloads/rhdp-downloads-app
19:14:40 01 10 2018 19:14:30.186:WARN [web-server]: 404: /@rhd/rhdp-downloads/rhdp-downloads-popular-product
19:14:40 01 10 2018 19:14:30.187:WARN [web-server]: 404: /@rhd/rhdp-downloads/rhdp-downloads-popular-products
19:14:40 01 10 2018 19:14:30.188:WARN [web-server]: 404: /@rhd/rhdp-downloads/rhdp-downloads-products
19:14:40 01 10 2018 19:14:30.188:WARN [web-server]: 404: /@rhd/rhdp-os-download
19:14:40 01 10 2018 19:14:30.188:WARN [web-server]: 404: /@rhd/rhdp-alert
19:14:40 01 10 2018 19:14:30.188:WARN [web-server]: 404: /@rhd/dp-stackoverflow
19:14:40 01 10 2018 19:14:30.189:WARN [web-server]: 404: /@rhd/rhdp-thankyou-page
19:14:40 01 10 2018 19:14:30.193:WARN [web-server]: 404: /@rhd/rhdp-tryitnow
19:14:40 01 10 2018 19:14:30.194:WARN [web-server]: 404: /@rhd/dp-referrer
```

**Comment by RubyGemTSL** [comment]
@LightGuard this is still a work in progress. @KamiQuasi do you need any support from me?

**Comment by LightGuard** [comment]
We're looking it today.

Jason Porter
Sent from Outlook<https://aka.ms/qtex0l> on iOS
________________________________

File: _tests/e2e/npm-debug.log [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+23 error If you do, this is most likely a problem with the developers.redhat.com package,
+23 error not with npm itself.
+23 error Tell the author that this fails on your system:
+23 error     wdio config/wdio.conf.docker.js "--baseUrl=https://developers-pr.stage.redhat.com/pr/2077/export/"
+23 error You can get information on how to open an issue for this project with:
+23 error     npm bugs developers.redhat.com
+23 error Or if that isn't available, you can get their info via:
+23 error     npm owner ls developers.redhat.com
+23 error There is likely additional logging output above.
+24 verbose exit [ 1, true ]
```
Comment (LightGuard) [inline _tests/e2e/npm-debug.log]: Please don't check this in

**Pull request merged**